content/docs/faq-state.md

```diff
@@ -1,106 +1,106 @@
 
-Here are some good resources for further reading on when to use `props` vs `state`:
+Ci-dessous quelques ressources utiles pour en apprendre plus à propos de quand utiliser `props` au lieu de `state`:
 * [Props vs State](https://github.com/uberVU/react-guide/blob/master/props-vs-state.md)
-* [ReactJS: Props vs. State](http://lucybain.com/blog/2016/react-state-vs-pros/)
+* [ReactJS : Props vs. State](http://lucybain.com/blog/2016/react-state-vs-pros/)
 
-### Why is `setState` giving me the wrong value? {#why-is-setstate-giving-me-the-wrong-value}
+### Pourquoi `setState` me retourne une mauvaise valeur ? {#why-is-setstate-giving-me-the-wrong-value}
 
-In React, both `this.props` and `this.state` represent the *rendered* values, i.e. what's currently on the screen.
+Dans React, `this.props` et `this.state` représente l'un et l'autre les valeurs rendues, i.e. ce qui se trouve actuellement à l'écran.
 
-Calls to `setState` are asynchronous - don't rely on `this.state` to reflect the new value immediately after calling `setState`. Pass an updater function instead of an object if you need to compute values based on the current state (see below for details).
+Les appels à `setState` sont asynchrones - ne comptez pas sur `this.state` pour refléter la nouvelle valeur immédiatement après avoir appelé `setState`. Passez une fonction de mise à jour à la place d'un objet si vous devez calculer des valeurs en fonction de l'état actuel (voir ci-dessous pour plus de détails).
 
-Example of code that will *not* behave as expected:
+Exemple de code qui ne se comporte *pas* comme attendu :
 
 ```jsx
 incrementCount() {
-  // Note: this will *not* work as intended.
+  // Note : cela ne va *pas* fonctionner comme prévu.
   this.setState({count: this.state.count + 1});
 }
 
 handleSomething() {
-  // Let's say `this.state.count` starts at 0.
+  // Disons que `this.state.count` commence à 0.
   this.incrementCount();
   this.incrementCount();
   this.incrementCount();
-  // When React re-renders the component, `this.state.count` will be 1, but you expected 3.
+  // Lorsque React refait le rendu du composant, `this.state.count` est à 1, pourtant, on espère avoir 3.
 
-  // This is because `incrementCount()` function above reads from `this.state.count`,
-  // but React doesn't update `this.state.count` until the component is re-rendered.
-  // So `incrementCount()` ends up reading `this.state.count` as 0 every time, and sets it to 1.
+  // C'est parce la fonction `incrementCount()` ci-dessus lit `this.state.count`,
+  // mais React ne met pas à jour `this.state.count` tant que le composant n'a pas refait le rendu.
+  // Donc `incrementCount()` finit par lire `this.state.count` à 0 à chaque fois et le définit à 1.
```

Comment:
meme si la traduction litterale est bonne je trouve cela un peut confus en francias (finit par lire)
on ne pourrait pas mettre a la place:  Donc `incrementCount()` lit `this.state.count`, qui est egal a 0 a chaque fois, et le definit a 1